File: js/bot.js

const discord = require("discord.js");
const fs = require("fs");

const NAME = "Erhardt";
const GENERAL_CHANNEL_ID = "XXX";
const ATTACHMENT_URL = "XXX";
const ADMIN_PWD = "...";

class JSDiscordBot extends discord.Client {
  constructor() {
    this.readConfig();
  }

  readConfig(){
    fs.readFile("../config.json", "utf8", (err, data) => {
      if (err) {
        this.printMessage("error", err);
      } else {
        that.config = JSON.parse(data);
      }
    });
  }

  printMessage(key, message){
    console.log(`[${new Date().now().toLocaleTimeString()}] [${key.toUpperCase()}] ${message}`);
  }

  run(){
    fs.readFile("../secret.json", "utf8", (err, data) => {
      if (err){
        printMessage("error", err)
      } else {
        that.login(JSON.parse(data).token);
      }
    });
  }
}

const client = new JSDiscordBot();

client.on("ready", () => {
  printMessage("LOGIN", "Logged in as " + client.user.tag);

  //client.user.setActivity("JS", {type: "PLAYING"});

  console.log("logged in on following channels:");
  client.guilds.forEach(guild => {
    guild.channels.forEach(channel => {
      console.log("guild: ${guild.name} - channel - ${channel.type} : ${channel.id}");
    });

  });

  const generalChannel = client.channels.get(GENERAL_CHANNEL_ID);
  //const attachment = new Discord.Attachment(ATTACHMENT_URL)
  generalChannel.send("Hey, I'm online!");
});

client.on("message", (message) => {
  // ignoring own messages
  if (message.author == client.user) return
  //msg.channel.send("I received your message: " + msg.content);

  if (message.content.toLowerCase().includes(NAME.toLowerCase())){
    // analyse message
    if (message.content.includes("$sudo") && message.content.includes("%" + ADMIN_PWD) && message.content.includes("shutdown"){
      process.exit(0);
    }
  }
});

client.run();

File: python/bot.py
#!/bin/python3

import discord
from datetime import datetime
import os; import sys
import pydiscbotutils as utils

class PythonDiscordBot(discord.Client):
    def __init__(self):
        super().__init__()
        self.config = utils.PyDiscBotConfig("main_channel", "bot_admin_channel", "Erhardt")

    # log in
    async def on_ready(self):
        self.print_message("login", "Logged in.")
        channel = client.get_channel(self.config.channels.bot_admin)
        await channel.send(self.config.name  + " got online.")
        return

    # handeling messages
    async def on_message(self, message):
        # ignoring own messages
        if message.author == client.user: return

        if lowercase(self.config.name) in lowercase(message.content):
            # the message is addressed to the bot ->
            # run message analysing code
            # runnning a command sent from bot admin
            if message.content == self.config.admin.shutdown_message:
                self.exit()
            return

    async def print_message(key, message):
        print(f"[{datetime.now().strftime('%Y.%m.%d %H:%M:%S')}] [{upper(key)}] {message}")
        return

    async def exit():
        await client.logout()
        sys.exit(0)

myBot = PythonDiscordBot()
try:
    myBot.run(utils.secret.token())
except FileNotFoundError:
    myBot.print_message("error", "You need to create a secret.json file.")
    myBot.print_message("info", "Structure of this file: {'token': 'YOUR_BOT_TOKEN', 'adminPwd': 'A_PASSWORD_TO_MANAGE_YOUR_BOT'}.")
